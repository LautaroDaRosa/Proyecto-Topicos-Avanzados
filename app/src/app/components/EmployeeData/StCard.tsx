import styled from 'styled-components';
import { media } from 'styles/media';

const StCard = styled.div`
  display: flex;
  border-radius: 20px;
  border: 1px solid var(--greyscale-colors-white, #fff);
  background: rgba(255, 255, 255, 0.25);
  backdrop-filter: blur(5px);
  padding: 20px;
  flex-direction: column;
  align-items: center;
  width: 100%;
  z-index: 2;

  ${media.medium} {
    gap: 15px;
  }

  ${media.large} {
    flex-direction: row;
    padding: 40px;
    gap: 25px;
  }
`;

export default StCard;
