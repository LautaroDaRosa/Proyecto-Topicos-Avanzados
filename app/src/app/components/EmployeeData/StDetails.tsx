import styled from 'styled-components';
import { media } from 'styles/media';

const StDetails = styled.div`
  gap: 15px;
  display: flex;
  flex-direction: column;
  align-items: center;

  ${media.medium} {
    align-items: center;
  }

  ${media.large} {
    flex-direction: row;
    width: 100%;
    justify-content: space-between;
  }
`;

export default StDetails;
