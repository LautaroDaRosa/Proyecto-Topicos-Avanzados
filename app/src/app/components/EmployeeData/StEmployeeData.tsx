import styled from 'styled-components';

const StEmployeeData = styled.div`
  display: flex;
  justify-content: center;
  align-items: center;
  position: relative;
  width: 100%;

  span {
    text-align: start;
    font-size: 16px;
    font-style: normal;
    font-weight: 400;
    line-height: normal;
  }
`;

export default StEmployeeData;
