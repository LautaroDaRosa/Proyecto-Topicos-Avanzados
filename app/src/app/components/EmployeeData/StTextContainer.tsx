import styled, { css } from 'styled-components';
import { media } from 'styles/media';

const StTextContainer = styled.div<{ justify: 'left' | 'right' }>`
  display: flex;
  box-shadow: none;
  padding: 0px;
  align-items: center;
  margin: 0px;
  flex-direction: column;

  h2 {
    font-size: 28px;
    font-style: normal;
    font-weight: 700;
    line-height: normal;
    text-transform: capitalize;
    margin: 0px;
  }

  ${props =>
    props.justify === 'left' &&
    css`
      ${media.large} {
        align-items: flex-start;
        gap: 10px;
      }
    `}

  ${props =>
    props.justify === 'right' &&
    css`
      ${media.large} {
        align-items: flex-end;
        gap: 10px;
      }
    `}
`;

export default StTextContainer;
