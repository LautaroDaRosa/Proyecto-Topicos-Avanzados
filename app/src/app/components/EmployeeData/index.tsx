import StCard from './StCard';
import StDetails from './StDetails';
import StEmployeeData from './StEmployeeData';
import StTextContainer from './StTextContainer';
import ProfileImage from '../ProfileImage';
import Ball from '../Ball';
import { getFormattedBirthday } from 'utils/dates';
import { getFormattedEntryDate } from 'utils/dates';
import { Employee } from 'types';

const EmployeeData = ({
  fullname,
  photo,
  dayOfBirth,
  city,
  entryDate,
}: Employee) => {
  return (
    <StEmployeeData>
      <StCard>
        <ProfileImage type="profile" src={photo} alt="User pic" />
        <StDetails>
          <StTextContainer justify="left">
            <h2>{fullname}</h2>
            <span>Working at 🐙 since: {getFormattedEntryDate(entryDate)}</span>
          </StTextContainer>
          <StTextContainer justify="right">
            <span>{'🎂 ' + getFormattedBirthday(dayOfBirth)}</span>
            <span>{'📍 ' + city}</span>
          </StTextContainer>
        </StDetails>
      </StCard>
      <Ball width={93} height={98} color="white" type="profile-ball" />
      <Ball width={78} height={84} color="blue" type="profile-ball" />
    </StEmployeeData>
  );
};

export default EmployeeData;
