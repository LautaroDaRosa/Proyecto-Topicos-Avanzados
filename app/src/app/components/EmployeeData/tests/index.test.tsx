import { render, screen } from '@testing-library/react';
import { BrowserRouter as Router } from 'react-router-dom';
import EmployeeData from '../index';

describe('Employee Data', () => {
  it('renders employee data correctly', () => {
    const employeeData = {
      fullname: 'Eliana Rosselli',
      photo: 'employees/erosselli.png',
      dob: '2020-07-02',
      city: 'Montevideo',
      entry_date: '2014-10-31',
    };

    const { container } = render(
      <Router>
        <EmployeeData
          fullname={employeeData.fullname}
          photo={employeeData.photo}
          dayOfBirth={employeeData.dob}
          city={employeeData.city}
          entryDate={employeeData.entry_date}
        />
      </Router>,
    );

    const userPic = screen.getByRole('img', { name: 'User pic' });
    screen.getByRole('heading', { name: 'Eliana Rosselli' });
    screen.getByText('🎂 02/07/2020');
    screen.getByText('📍 Montevideo');
    screen.getByText('Working at 🐙 since: 31 October 2014');

    expect(userPic).toHaveAttribute('src', 'employees/erosselli.png');
    expect(container).toMatchSnapshot();
  });
});
