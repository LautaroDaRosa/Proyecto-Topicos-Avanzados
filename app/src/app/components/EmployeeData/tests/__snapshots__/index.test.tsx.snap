// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Employee Data renders employee data correctly 1`] = `
.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  border-radius: 20px;
  border: 1px solid var(--greyscale-colors-white,#fff);
  background: rgba(255,255,255,0.25);
  -webkit-backdrop-filter: blur(5px);
  backdrop-filter: blur(5px);
  padding: 20px;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 100%;
  z-index: 2;
}

.c3 {
  gap: 15px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  position: relative;
  width: 100%;
}

.c0 span {
  text-align: start;
  font-size: 16px;
  font-style: normal;
  font-weight: 400;
  line-height: normal;
}

.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-shadow: none;
  padding: 0px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin: 0px;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c4 h2 {
  font-size: 28px;
  font-style: normal;
  font-weight: 700;
  line-height: normal;
  text-transform: capitalize;
  margin: 0px;
}

.c5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-shadow: none;
  padding: 0px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin: 0px;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c5 h2 {
  font-size: 28px;
  font-style: normal;
  font-weight: 700;
  line-height: normal;
  text-transform: capitalize;
  margin: 0px;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  object-fit: cover;
  height: 122px;
  width: 122px;
  border-radius: 350px;
}

.c6 {
  position: absolute;
  height: 98px;
  width: 93px;
  z-index: -1;
  right: 5px;
  top: -50px;
}

.c7 {
  position: absolute;
  height: 84px;
  width: 78px;
  z-index: -1;
  right: 32px;
  bottom: -32px;
}

@media (min-width:768px) {
  .c1 {
    gap: 15px;
  }
}

@media (min-width:1280px) {
  .c1 {
    -webkit-flex-direction: row;
    -ms-flex-direction: row;
    flex-direction: row;
    padding: 40px;
    gap: 25px;
  }
}

@media (min-width:768px) {
  .c3 {
    -webkit-align-items: center;
    -webkit-box-align: center;
    -ms-flex-align: center;
    align-items: center;
  }
}

@media (min-width:1280px) {
  .c3 {
    -webkit-flex-direction: row;
    -ms-flex-direction: row;
    flex-direction: row;
    width: 100%;
    -webkit-box-pack: justify;
    -webkit-justify-content: space-between;
    -ms-flex-pack: justify;
    justify-content: space-between;
  }
}

@media (min-width:1280px) {
  .c4 {
    -webkit-align-items: flex-start;
    -webkit-box-align: flex-start;
    -ms-flex-align: flex-start;
    align-items: flex-start;
    gap: 10px;
  }
}

@media (min-width:1280px) {
  .c5 {
    -webkit-align-items: flex-end;
    -webkit-box-align: flex-end;
    -ms-flex-align: flex-end;
    align-items: flex-end;
    gap: 10px;
  }
}

@media (min-width:768px) {
  .c6 {
    bottom: -35px;
    right: 20px;
    top: auto;
  }
}

@media (min-width:1280px) {
  .c6 {
    bottom: -24px;
    left: -60px;
  }
}

@media (min-width:768px) {
  .c7 {
    left: 20px;
    top: -25px;
  }
}

@media (min-width:1280px) {
  .c7 {
    left: auto;
    top: -15px;
    right: -55px;
  }
}

<div>
  <div
    class="c0"
  >
    <div
      class="c1"
    >
      <img
        alt="User pic"
        class="c2"
        src="employees/erosselli.png"
        type="profile"
      />
      <div
        class="c3"
      >
        <div
          class="c4"
        >
          <h2>
            Eliana Rosselli
          </h2>
          <span>
            Working at üêô since: 
            31 October 2014
          </span>
        </div>
        <div
          class="c5"
        >
          <span>
            üéÇ 02/07/2020
          </span>
          <span>
            üìç Montevideo
          </span>
        </div>
      </div>
    </div>
    <img
      alt="Ball"
      class="c6"
      color="white"
      height="98"
      src="white-ball.svg"
      type="profile-ball"
      width="93"
    />
    <img
      alt="Ball"
      class="c7"
      color="blue"
      height="84"
      src="blue-ball.svg"
      type="profile-ball"
      width="78"
    />
  </div>
</div>
`;
